tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -52,17 +57,32 @@ def get_storage_account_connection_string(id):
         return connection_string
 
     @staticmethod
-    def deploy_dedicated_function_app(parameters):
+    def is_consumption_plan(function_params):
+        return StringUtils.equal(function_params.service_plan['sku_tier'], 'dynamic')
+
+    @staticmethod
+    def deploy_function_app(parameters):
         function_app_unit = FunctionAppDeploymentUnit()
         function_app_params = \
             {'name': parameters.function_app_name,
              'resource_group_name': parameters.function_app_resource_group_name}
         function_app = function_app_unit.get(function_app_params)
+
         if function_app:
+            # retrieve the type of app service plan hosting the existing function app
+            session = local_session(Session)
+            web_client = session.client('azure.mgmt.web.WebSiteManagementClient')
+            app_id = function_app.server_farm_id
+            app_name = ResourceIdParser.get_resource_name(app_id)
+            app_resource_group_name = ResourceIdParser.get_resource_group(app_id)
+            app_service_plan = web_client.app_service_plans.get(app_resource_group_name, app_name)
+            parameters.service_plan['sku_tier'] = app_service_plan.sku.tier
```

Comment:
Does it make sense to return sku.tier from the function to avoid silent modification of parameters?

Reply:
I added a comment.